crates/telio-utils/src/git.rs

```diff
@@ -4,11 +4,8 @@ pub fn commit_sha() -> &'static str {
     option_env!("LIBTELIO_COMMIT_SHA").unwrap_or("dev")
 }
 
-/// Get the version tag that was used when building.
-/// If not present defaults to "dev".
+/// Get the version placeholder (half of the maximum length of git tag)
+/// Will be replaced during build promotions
 pub fn version_tag() -> &'static str {
-    option_env!("LIBTELIO_COMMIT_TAG")
-        .map(|tag| tag.trim())
-        .and_then(|tag| if tag.is_empty() { None } else { Some(tag) })
-        .unwrap_or("dev")
+    "VERSION_PLACEHOLDER@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@"
```

Comment:
Shouldn't this be some sort of `CStr`? Normal `&str` in not null terminated so will have a length independent of the position of first null byte: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a288ca452d106cf2604838c4610382df

Reply:
fixed I guess :smile:

Reply:
I'm wondering if we shouldn't test this somehow. Maybe create an [example executable](https://doc.rust-lang.org/cargo/reference/cargo-targets.html#examples) that all it does is output the result of `version_tag` to stdout, apply modification to it using the script and verify that the output has changed? This could be run on mac,win,linux on github CI to make sure that we aren't breaking the libraries somehow.